test/e2e/static_pods.go

```diff
@@ -0,0 +1,100 @@
+package e2e
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	"github.com/onsi/ginkgo"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/kubernetes/test/e2e/framework"
+	e2enode "k8s.io/kubernetes/test/e2e/framework/node"
+	e2epod "k8s.io/kubernetes/test/e2e/framework/pod"
+)
+
+func createStaticPod(f *framework.Framework, nodeName string, podYaml string) {
+	//create file
+	var podFile = "static-pod.yaml"
+	if err := ioutil.WriteFile(podFile, []byte(podYaml), 0644); err != nil {
+		framework.Failf("Unable to write static-pod.yaml  to disk: %v", err)
+	}
+	defer func() {
+		if err := os.Remove(podFile); err != nil {
+			framework.Logf("Unable to remove the static-pod.yaml from disk: %v", err)
+		}
+	}()
+	var dst = fmt.Sprintf("%s:/etc/kubernetes/manifests/%s", nodeName, podFile)
+	cmd := []string{"docker", "cp", podFile, dst}
```

Comment:
This is.... a possible edge case, that I'd consider optional to handle, but...

In the case that the kubelet starts creating the pod before all the bytes are written to the file -- it's possible it'll kick it off with only part of the file, and the pod will fail to come up, or come up in a failed state. You can avoid this by writing the file to a temporary location, e.g. `/etc/my.temp.manifests/static-pod.yml` and the moving it into the final destination a la `mv /etc/my.temp.manifests/static-pod.yml /etc/kubernetes/manifests/static-pod.yml` -- then it's an atomic operation.

IMO, I'd give it a half dozen rounds of testing and see if you hit this situation. But, you could avoid it entirely by doing the move for the atomic operation.

Reply:
Ben and I chatted it out, and Ben didn't hit it after a number of tests. So, I think we're OK with the atomic operation. Thanks Ben!

Reply:
Valid point @dougbtv and I agree we may not need to worry about it here. The reason being is that the pod creation does not really happen while file is written, but only after the `docker cp` command takes place.